### PR TITLE
refactor: :truck: Moved methods handling mod state

### LINKED ISSUE #281: Move mod handling functions out of `ModLoaderMod` into a new class `ModLoaderModManager`
> it's no longer only about building mods since we have added `disable_mods` and the others. now that I say it out loud, do we need another split into a class that handles the mod states? everything that mod developers don't need to touch, but that game developers need to manage the mods?

_Originally posted by @Qubus0 in https://github.com/GodotModding/godot-mod-loader/pull/279#discussion_r1232292063_

### PR DESCRIPTION
**Pull request by KANAjetzt**
Moved the methods handling mod state from `ModLoaderMod` to `ModLoaderModManager`. Additionally, all getter methods inside `ModLoaderMod` have been moved together at the bottom.

closes #281

### CODE REVIEW

**Review by Qubus0 — APPROVED** [review]
Neat

File: addons/mod_loader/api/mod_manager.gd [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# This Class provides methods to manage mod state.
+# *Note: Intended to be used by game developers.*
+class_name ModLoaderModManager
```
Comment (ithinkandicode) [inline addons/mod_loader/api/mod_manager.gd:3]: Does the new class need to be added to `new_global_classes` in *mod_loader_setup.gd*?

Comment (Qubus0) [reply addons/mod_loader/api/mod_manager.gd:3]: right, yes. i always forget about that.

Comment (KANAjetzt) [reply addons/mod_loader/api/mod_manager.gd:3]: ty - fixed in [10025ca](https://github.com/GodotModding/godot-mod-loader/pull/283/commits/10025ca3293aee13155afb67d94d6d004f4528bc) 👍

**Pull request merged**